modules/users/server/routes/admin.server.routes.js

```diff
@@ -4,22 +4,28 @@
  * Module dependencies
  */
 var adminPolicy = require('../policies/admin.server.policy'),
-  admin = require('../controllers/admin.server.controller');
+  admin = require('../controllers/admin.server.controller'),
+  passport = require('passport'),
+  express = require('express');
 
 module.exports = function (app) {
-  // User route registration first. Ref: #713
-  require('./users.server.routes.js')(app);
+  var router = express.Router();
 
-  // Users collection routes
-  app.route('/api/users')
+  //Set JWT Auth for all Admin Routes
+  router.all('*', passport.authenticate('jwt', { session: false }));
+
+  router.route('/')
     .get(adminPolicy.isAllowed, admin.list);
 
   // Single user routes
-  app.route('/api/users/:userId')
+  router.route('/:userId')
     .get(adminPolicy.isAllowed, admin.read)
     .put(adminPolicy.isAllowed, admin.update)
     .delete(adminPolicy.isAllowed, admin.delete);
 
   // Finish by binding the user middleware
-  app.param('userId', admin.userByID);
+  router.param('userId', admin.userByID);
+
+  app.use('/api/admin/users', router);
```

Comment:
Possibly but this is confusing to me:
```js
'use strict';

var passport = require('passport'),
  express = require('express'),
  adminPolicy = require('../policies/admin.server.policy'),
  admin = require('../controllers/admin.server.controller'),
  users = require('../controllers/users.server.controller');

module.exports = function (app) {
  // Create Router
  var router = express.Router();

  //Set JWT Auth for all user Routes
  router.all('*', passport.authenticate('jwt', { session: false }));

  // Setting up the users profile api
  router.route('/me').get(users.me);
  router.route('/')
    .get(adminPolicy.isAllowed, admin.list);
    .put(users.update);
  router.route('/accounts').delete(users.removeOAuthProvider);
  router.route('/password').post(users.changePassword);
  router.route('/picture').post(users.changeProfilePicture);

  //Admin Routes
  router.route('/:userId')
    .get(adminPolicy.isAllowed, admin.read)
    .put(adminPolicy.isAllowed, admin.update)
    .delete(adminPolicy.isAllowed, admin.delete);

  // Finish by binding the user middleware
  router.param('userId', admin.userByID);

  app.use('/api/users', router);
};
```

so api/users PUT a user can access but GET only and admin can.  To me it is bad to mix admin and user routes.  Seems to me like someone could confuse this, or even worse change something that allows non admins to access user data.